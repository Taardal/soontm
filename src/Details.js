import React from "react";
import PropTypes from "prop-types";
import { View, Text, Image } from "react-native";
import { styles } from "./Details.styles";

const WIDTH = 110;
const HEIGHT = (WIDTH * 5) / 3;

class Details extends React.Component {
  render() {
    const { movie, posterUrl, backdropUrl, language } = this.props;
    return (
      <View style={styles.container}>
        <View style={{flex: 3}}>
          <Image source={{ uri: backdropUrl }} style={styles.backdrop} />
          <View style={styles.header} />
          <View style={[styles.headerOverlay, { height: HEIGHT }]}>
          <Image
            source={{ uri: posterUrl }}
            style={[
              styles.poster,
              {
                width: WIDTH,
                height: HEIGHT,
              }
            ]}
          />
          <View style={styles.infoFrame}>
            <View style={styles.infoText}>
              <Text style={styles.title}>{movie.title}</Text>
              <View style={styles.subtitleText}>
                <Text style={styles.releaseDate}>{movie.release_date}</Text>
                <Text style={styles.language}>{language}</Text>
              </View>
            </View>
          </View>
        </View>
        </View>
        <View style={styles.overview}>
          <View style={styles.overviewText}>
            <Text style={styles.overviewTitleText}>Overview</Text>
            <Text>{movie.overview}</Text>
          </View>
        </View>
      </View>
    );
  }
}

Details.propTypes = {
  movie: PropTypes.object.isRequired,
  posterUrl: PropTypes.string.isRequired,
  backdropUrl: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired
}

export default Details;
