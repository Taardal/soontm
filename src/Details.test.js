import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Enzyme from "enzyme";
import Details from "./Details";

jest.mock("Text", () => "Text");

describe("Details", () => {
  it("matches snapshot", () => {
    const movie = {
      id: 1,
      title: "movie",
      release_date: "release_date",
      original_language: "en",
      overview: "overview"
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Details movie={movie} posterUrl={"posterUrl"} backdropUrl={"backdropUrl"} language={"English"} />);
    const renderOutput = shallowRenderer.getRenderOutput();

    expect(renderOutput).toMatchSnapshot();
  });
});
