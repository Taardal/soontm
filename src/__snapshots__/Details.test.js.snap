// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details matches snapshot 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "backdropUrl",
        }
      }
      style={
        Object {
          "flex": 2,
        }
      }
    />
    <Component
      style={
        Object {
          "backgroundColor": "#212121",
          "flex": 1,
        }
      }
    />
    <Component
      style={
        Array [
          Object {
            "bottom": 20,
            "flex": 1,
            "flexDirection": "row",
            "left": 25,
            "position": "absolute",
            "right": 25,
          },
          Object {
            "height": 183.33333333333334,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "posterUrl",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "resizeMode": "cover",
            },
            Object {
              "height": 183.33333333333334,
              "width": 110,
            },
          ]
        }
      />
      <Component
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Component
          style={
            Object {
              "marginLeft": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            movie
          </Text>
          <Component
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                }
              }
            >
              release_date
            </Text>
            <Text
              style={
                Object {
                  "color": "white",
                  "marginLeft": 10,
                }
              }
            >
              English
            </Text>
          </Component>
        </Component>
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Component
      style={
        Object {
          "marginBottom": 35,
          "marginLeft": 25,
          "marginRight": 25,
          "marginTop": 35,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Overview
      </Text>
      <Text>
        overview
      </Text>
    </Component>
  </Component>
</Component>
`;
